### PR TITLE
add ability to name the value field

### PR DESCRIPTION
**Pull request by jessingrass**
### Required for all PRs:

- [ ] Signed [CLA](https://influxdata.com/community/cla/).
- [ ] Associated README.md updated.
- [ ] Has appropriate unit tests.

### CODE REVIEW

File: config/config.go [hunk 1]
```diff
@@ -1323,6 +1323,9 @@ func (c *Config) getParserConfig(name string, tbl *ast.Table) (*parsers.Config,
 
 	c.getFieldStringSlice(tbl, "form_urlencoded_tag_keys", &pc.FormUrlencodedTagKeys)
 
+	//for value parser
+	c.getFieldString(tbl, "value_field", &pc.ValueValueName)
```
Comment (ssoroka) [inline config/config.go:1327]:
```suggestion
	c.getFieldString(tbl, "value_value_field", &pc.ValueValueField)
```

File: plugins/parsers/registry.go [hunk 2]
```diff
@@ -148,6 +148,9 @@ type Config struct {
 
 	// FormData configuration
 	FormUrlencodedTagKeys []string `toml:"form_urlencoded_tag_keys"`
+
+	// Custom Value name for Value field
+	ValueValueName string `toml:"value_value_name"`
```
Comment (ssoroka) [inline plugins/parsers/registry.go:153]:
```suggestion
	ValueValueField string `toml:"value_value_field"`
```

File: plugins/parsers/registry.go [hunk 3]
```diff
@@ -172,7 +175,7 @@ func NewParser(config *Config) (Parser, error) {
 		)
 	case "value":
 		parser, err = NewValueParser(config.MetricName,
-			config.DataType, config.DefaultTags)
+			config.DataType, config.DefaultTags, config.ValueValueName)
```
Comment (ssoroka) [inline plugins/parsers/registry.go:178]:
```suggestion
			config.DataType, config.DefaultTags, config.ValueValueField)
```

**Review by ssoroka — CHANGES_REQUESTED** [review]
looks like I gave you some conflicting naming advice. comments!

**Comment by srebhan** [comment]
What are you trying to achive here??? If you want to override a field-name maybe `fieldname_override` similar to `name_override`?

**Comment by jessingrass** [comment]
@srebhan issue 7702 (linked above) is the issue I'm working to address

**Review by ssoroka — APPROVED** [review]

**Comment by ssoroka** [comment]
oh. you'll also need to default the value field name to "value"

File: plugins/parsers/registry.go [hunk 4]
```diff
@@ -148,6 +148,9 @@ type Config struct {
 
 	// FormData configuration
 	FormUrlencodedTagKeys []string `toml:"form_urlencoded_tag_keys"`
+
+	// Custom Value name for Value field
+	ValueValueField string `toml:"value_value_field"`
```
Comment (srebhan) [inline plugins/parsers/registry.go:153]: This is a horrible user-facing option name! How about `value_fieldname`?

File: plugins/parsers/value/parser.go [hunk 5]
```diff
@@ -46,7 +47,7 @@ func (v *ValueParser) Parse(buf []byte) ([]telegraf.Metric, error) {
 		return nil, err
 	}
 
-	fields := map[string]interface{}{"value": value}
+	fields := map[string]interface{}{v.ValueField: value}
```
Comment (srebhan) [inline plugins/parsers/value/parser.go:50]: You need to handle the default i.e. the user didn't specify anything...
```suggestion
	fieldname := "value"
	if v.ValueField != "" {
		fieldname = v.ValueField
	}
	fields := map[string]interface{}{fieldname: value}
```

**Review by srebhan — CHANGES_REQUESTED** [review]
You need to handle the default-case where the user didn't specify anything. Furthermore, you should update the README.md to introduce the new option and give a short explanation of what it does. Last but not least I really think `value_value_field` is a horrible name for an option and suggest changing it to `value_fieldname`.

**Comment by srebhan** [comment]
@jessingrass any news on this one?

**Review by srebhan — APPROVED** [review]
Looks good to me!

**Review by helenosheaa — APPROVED** [review]
Looks good!

**Comment by srebhan** [comment]
@jessingrass can you please resolve the merge conflicts!

**Comment by Jagularr** [comment]
The fix for this has been made in #8979 when we had the related issue assigned for pairing yesterday. I did not realize there was already a PR out for this, my apologies!!!

**Pull request closed without merging**